src/main/java/org/haffson/adventofcode/days/day03/Day03.java

```diff
@@ -86,4 +80,37 @@ List<RectangleClaim> convertStringToRectangleList(final List<String> stringList)
 
         return stringList.stream().map(RectangleClaim::of).collect(Collectors.toList());
     }
+
+    private int calculateNotOverlappingClaim(final List<String> inputList) {
+        final List<RectangleClaim> rectangleClaimList = convertStringToRectangleList(inputList);
+        final String[][] fabric = get2DClaimArray(rectangleClaimList);
+
+        final int[] intactClaimId = {-1};
+        rectangleClaimList .forEach(claim ->{
+            boolean stillFree = true;
+            for (int x = claim.getLeftSpace(); x < claim.getRightXCoordinate() && stillFree; x++){
+                for (int y = claim.getTopSpace(); y < claim.getBottomYCoordinate(); y++) {
+                    if (!Objects.equals(fabric[x][y], claim.getId())){
+                        stillFree = false;
+                        break;
+                    }else if (x == claim.getRightXCoordinate() - 1 && y == claim.getBottomYCoordinate() - 1){
+                        intactClaimId[0] = Integer.parseInt(claim.getId());
+                    }
+                }
+            }
+        });
+        return intactClaimId[0];
+    }
+
+    private String[][] get2DClaimArray(final List<RectangleClaim> claimList){
```

Comment:
What about adding more tests to test the edge cases? I mean also the not "happy case". Think about how to secure your implementation, e.g. when someone joins your project and starts to refactor your method. What can you think of?